go/arrow/flight/record_batch_reader.go

```diff
@@ -60,15 +68,41 @@ func (d *dataMessageReader) Release() {
 		if d.msg != nil {
 			d.msg.Release()
 			d.msg = nil
+			d.lastAppMetadata = nil
 		}
 	}
 }
 
+type Reader struct {
+	*ipc.Reader
+	dmr *dataMessageReader
+}
+
+func (r *Reader) Retain() {
```

Comment:
here and below, these are public methods?  It would be nice to add some documentation.

Reply:
added